src/libraries/Microsoft.Extensions.DependencyInjection/src/ServiceLookup/CallSiteFactory.cs

```diff
@@ -614,6 +686,20 @@ public bool IsService(Type serviceType)
                    serviceType == typeof(IServiceProviderIsService);
         }
 
+        /// <summary>
+        /// Returns true if both keys are null or equals, or if key1 is KeyedService.AnyKey and key2 is not null
+        /// </summary>
+        private static bool KeyEquals(object? key1, object? key2)
```

Comment:
nit: maybe use "match" terminology instead of "equals" and perhaps rename the arguments to `key` and `query` to convey the meaning a little more clearly

Reply:
Agree